ledger/participant-integration-api/src/main/scala/platform/store/interning/StringInterningCache.scala

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.interning
+
+import com.daml.dec.DirectExecutionContext
+import com.daml.lf.data.Ref
+import com.daml.logging.LoggingContext
+
+import scala.concurrent.Future
+
+class DomainStringIterators(
+    val parties: Iterator[String],
+    val templateIds: Iterator[String],
+)
+
+trait StringInterningCache extends StringInterning {
```

Comment:
As discussed offline, `View` would better suit the name of this structure, since it's a projection of the index.

Reply:
Yeah I think so too. @meiersi-da are you ok with this naming? (in the design doc we use the term Covering Cache)